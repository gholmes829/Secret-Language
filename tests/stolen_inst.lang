# stealing init should return this

class Foo {
    void init() {  # TODO change fn signature so init returns Foo
        print(this)
    }
}

num main() {
    let foo = Foo()
    print(foo.init())
    return 0
}