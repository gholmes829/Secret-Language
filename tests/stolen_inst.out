<Internal Instance of "Foo">
<Internal Instance of "Foo">
<Internal Instance of "Foo">
